source/01-physics-prototype/actors/Dash.gd

```diff
@@ -32,20 +31,27 @@ func check_walls():
 
 
 func stop():
+	if not platform_actor.is_on_floor():
+		air_dash_available = false
 	emit_signal("finished")
-	set_physics_process(false)
+	set_enabled(false)
 
 
-func set_physics_process(enable):
+func set_enabled(enable):
 	if enable:
 		if not platform_actor.is_on_floor() and not air_dash:
 			return
+		if not platform_actor.is_on_floor() and air_dash:
+			if not air_dash_available:
+				return
 		emit_signal("started")
 		initial_position = platform_actor.position
 	else:
 		emit_signal("finished")
-	.set_physics_process(enable)
+	set_physics_process(enable)
 
 
-func set_air_dash(enabled):
-	air_dash = enabled
+func set_air_dash(enable):
+	air_dash = enable
+	if enable:
+		air_dash_available = true
```

Comment:
This is kinda confusing, sounds redundant. If the `air_dash` is enabled we also make it available? Isn't this implicit some how?